org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/RequestManager.java

```diff
@@ -55,25 +53,17 @@ public void shutdown() {
 		parallel.shutdown();
 		cancel();
 	}
-	
```

Comment:
@szarnekow now do we deal with these removals api wise?

Reply:
I'm wondering the same

Reply:
Question: Do you think it'd help to keep the overloaded constructors in the ReadRequest and WriteRequest with a RequestManager instead of an AbstractRequestManager?

Reply:
Yes. I think we need to make a judgement call here. I believe it's ok to break binary compatibility for the hierarchy around AbstractRequest

Reply:
Are you any further with this?

Reply:
We had to postpone the committer call to Friday this week. I'll get back to you

Reply:
We are fine with this binary incompatibility. Adopters will need to make a minor adjustment to their code if they are already using a custom implementation in rare cases. That’s ok.

Reply:
One point remaining about the Guice bindings.